src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -33,7 +33,7 @@ trait IterableLike[+A, +C[X] <: Iterable[X]]
   /** Create a collection of type `C[A]` from the elements of `coll`, which has
     *  the same element type as this collection. Overridden in StringOps and ArrayOps.
     */
-  protected[this] def fromIterableWithSameElemType(coll: Iterable[A]): C[A] = fromIterable(coll)
+  protected[this] def fromIterableWithSameElemType(coll: Iterable[A]): C[A]
```

Comment:
Hm, I just ran into two different compiler bugs (one probably being https://issues.scala-lang.org/browse/SI-10081) while trying this out. We're pushing the limits here...